### PR TITLE
[BUG] Invalid copy into descendant in different spaces

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App: https://github.com/owncloud/android/issues/4293

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

**Review by JuancaG05 — CHANGES_REQUESTED** [review]

File: owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/CopyFileUseCase.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ class CopyFileUseCase(
     fun validateOrThrowException(listOfFilesToCopy: List<OCFile>, targetFolder: OCFile) {
         require(listOfFilesToCopy.isNotEmpty())
 
-        if (listOfFilesToCopy.any { targetFolder.remotePath.startsWith(it.remotePath) }) {
+        if (listOfFilesToCopy.any { targetFolder.remotePath.startsWith(it.remotePath) and (targetFolder.spaceId == it.spaceId) }) {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/CopyFileUseCase.kt]: Changed.

File: changelog/unreleased/4295 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Copy folder to another space correctly
+
+Copying a folder to a space inside another folder with the same name as the folder
+you want to copy works correctly.
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply changelog/unreleased/4295]: Changed.

File: changelog/unreleased/4295 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Copy folder to another space correctly
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply changelog/unreleased/4295]: Changed.

**Review by JuancaG05 — APPROVED** [review]
LGTM now 👍

**Comment by jesmrec** [comment]
Fixed.

No weird cases were found.

**Pull request merged**